### ISSUE TITLE
ZipFile can only be used ... Issue with userpoolclient and MFA lambda

### ISSUE THREAD

**Issue by mwarger** [aws-amplify/amplify-cli#2514]
**Describe the bug**
I'm getting the following error when pushing to the amplify console in my development environment:
ZipFile can only be used when Runtime is set to either of nodejs, nodejs4.3, nodejs6.10, nodejs8.10, python2.7, python3.6, python3.7.

I'm getting this for the userpoolclient lambda and the MFA lambda.

**To Reproduce**
1. Follow migration steps to update 6.10 to 8.10

Please let me know what information would be helpful.  

**Expected behavior**
I would expect this to update successfully

**Comment by yuth** [comment]
@mwarger what was the version of Amplify CLI that was used to provision the userpool?

**Comment by mwarger** [comment]
@yuth you mean originally?  or the version of the CLI that i'm running?  or the version of the amplify console's CLI?

**Comment by yuth** [comment]
You mentioned 
> Follow migration steps to update 6.10 to 8.10

So did add auth using a older version of CLI? Also whats the version of CLI you have installed?

Could you also include the changes you have made in the CloudFormation templates to migrate

**Comment by mwarger** [comment]
My amplify version - amplify -v
3.11.0

I added auth with a previous version of the CLI.  

I followed the instructions after I noticed that my project was initially failing because of an outdated node version: https://aws-amplify.github.io/docs/cli/lambda-node-version-update#node-version-update.

From the console logs:
The following resource(s) failed to update: [UserPoolClientLambda, MFALambda].

**Comment by yuth** [comment]
@mwarger what value do you have for `Runtime` for both `UserPoolClientLambda` and `MFALambda`?

**Comment by mwarger** [comment]
nodejs8.10

**Comment by yuth** [comment]
Thanks @mwarger, could you share the complete cloudformation logs that gets printed when you push the change

**Comment by mwarger** [comment]
@yuth Where can I send it?

**Comment by yuth** [comment]
amplify-cli@amazon.com

**Comment by mwarger** [comment]
> [amplify-cli@amazon.com](mailto:amplify-cli@amazon.com)

@yuth - Will send that later this evening.  Thank you.

**Comment by mwarger** [comment]
@yuth - Sent.

**Comment by UnleashedMind** [comment]
@mwarger 
The same type of error happened with nodejs version updates, see e.g.
https://forums.aws.amazon.com/thread.jspa?messageID=908801

But nodejs8.10 is supported by CloudFormation now, and we didn't see that error when we were testing our migration. 
I've contacted the Cloudfomration service team to investigate further, will keep you updated.

**Comment by mwarger** [comment]
@UnleashedMind Thanks.  I have been communicating with @yuth as well as opened a ticket through AWS.  I was asked about the same upgrade issues (which I have already done).  Hopefully, I can resolve this soon.

**Comment by kaustavghosh06** [comment]
@mwarger Were you able to unblock yourself on this?

**Comment by mwarger** [comment]
@kaustavghosh06 I'm still in contact with AWS support about this.  Apparently someone on the AppSync team is talking with cloudformation team (I think it's @mikeparisstuff).  Will keep you posted.

**Comment by mwarger** [comment]
This has been resolved through some walkthrough with support.  My cloudformation stack was left in a status `UPDATE_ROLLBACK_FAILED` and to resolve this issue I had to attempt to continue the rollback using the console and choose to skip rolling back the 3 affected lambda functions.  My amplify project was able to roll back successfully and then I was able to re-deploy and resolve the issue.
